lib/awspec/type/transit_gateway.rb

```diff
@@ -18,9 +18,21 @@ def options
     end
 
     def has_attachment?(att_id)
-      atts = find_tgw_attachments_by_tgw_id(@id)
-      ret = atts.find_all { |att| att.transit_gateway_attachment_id == att_id }
-      ret.any?
+      atts = find_tgw_attachments_by_tgw_id(id)
```

Comment:
Specific bugfix on this line changing `@id` to `id` as `@id` wasn't initialised if `should exist` hadn't been called first.